---
title: "[MapReduce] in state wait_for_input terminated with reason:	processing_error"
description: ""
project: community
lastmod: 2012-09-25T04:19:39-07:00
sitemap:
  priority: 0.2
layout: mailinglistitem
mailinglist_id: "msg08690"
author_name: "郎咸武"
project_section: "mailinglistitem"
sent_date: 2012-09-25T04:19:39-07:00

---


Hi guys, I need yours help.
I using mapreduce in my server. I only query from riak, that is ok. But
first delete some data,after query is error in the follow.

My environment:
Linux jason-lxw 3.2.0-30-generic #48-Ubuntu SMP Fri Aug 24 16:52:48 UTC
2012 x86\_64 x86\_64 x86\_64 GNU/Linux
Erlang R14B04
riak-1.2
riak-erlang-1.2

Thanks in advance.
Jason

My server error message:
=ERROR REPORT==== 25-Sep-2012::18:30:48 ===
E(<0.411.0>:mod\_pubsub\_riak:3168) : In mod\_pubsub\_riak:get\_newsfeed 3168
Erro={error,

 {0,


<<"{\"phase\":0,\"error\":\"badarg\",\"input\":\"{ok,{r\_object,<<\\\"itrends/ejabberd/pubsub\_item\\\">>,<<131,104,2,107,0,13,53,52,52,56,56,48,70,56,66,52,54,53,48,97,5>>,[{r\_content,{dict,4,16,16,8,80,48,{[],[],[],[],[],[],[],[],[],[],[],[],[],[],[],[]},{{[],[],[],[],[],[],[],[],[],[],[[<<\\\"X-Riak-VTag\\\">>,51,100,111,68,108,85,106,107,48,88,110,120,51,79,106,56,71,68,97,87,74,54]],[[<<\\\"index\\\">>]],[[<<\\\"X-Riak-Deleted\\\">>,116,114,117,101]],[[<<\\\"X-Riak-Last-Modified\\\">>|{1348,569048,516672}]],[],[]}}},<<>>}],[{<<35,9,254,249,80,93,197,65>>,{2,63515788248}}],...},...}\",\"type\":\"error\",\"stack\":\"[{erlang,binary\_to\_term,[<<>>]},{trend\_riak,get\_map\_bucket\_values,3},{riak\_kv\_mrc\_map,map,3},{riak\_kv\_mrc\_map,process,3},{riak\_pipe\_vnode\_worker,process\_input,3},{riak\_pipe\_vnode\_worker,wait\_for\_input,2},{gen\_fsm,handle\_msg,7},{proc\_lib,init\_p\_do\_apply,3}]\"}">>}}


Riak error in the console:

 undefined}, Nodes=["jason1"],
CategoryList=[]
16:24:01.346 [error] gen\_fsm <0.2681.0> in state wait\_for\_input terminated
with reason: processing\_error
16:24:01.346 [error] Pipe worker startup failed:

 
exit:{shutdown,{gen\_fsm,sync\_send\_event,[<0.2654.0>,{get\_details,#Ref<0.0.0.8144>,456719261665907161938651510223838443642478919680,<0.327.0>}]}}

 
[{gen\_fsm,sync\_send\_event,2},{riak\_pipe\_fitting,get\_details,2},{riak\_pipe\_vnode,new\_worker,2},{riak\_pipe\_vnode,enqueue\_internal,3},{riak\_core\_vnode,vnode\_command,3},{gen\_fsm,handle\_msg,7},{proc\_lib,init\_p\_do\_apply,3}]
16:24:01.346 [error] Pipe worker startup failed:

 
exit:{shutdown,{gen\_fsm,sync\_send\_event,[<0.2654.0>,{get\_details,#Ref<0.0.0.8144>,799258707915337533392640142891717276374338109440,<0.357.0>}]}}

 
[{gen\_fsm,sync\_send\_event,2},{riak\_pipe\_fitting,get\_details,2},{riak\_pipe\_vnode,new\_worker,2},{riak\_pipe\_vnode,enqueue\_internal,3},{riak\_core\_vnode,vnode\_command,3},{gen\_fsm,handle\_msg,7},{proc\_lib,init\_p\_do\_apply,3}]
16:24:01.347 [error] Pipe worker startup failed:fitting was gone before
startup
16:24:01.348 [error] Pipe worker startup failed:fitting was gone before
startup
16:24:01.389 [error] CRASH REPORT Process <0.2681.0> with 0 neighbours
exited with reason: processing\_error in gen\_fsm:terminate/7
16:24:01.408 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,{kvget\_map,0},{trace,[error],{vnode,{fitting\_died,707914855582156101004909840846949587645842325504}}}}
16:24:01.545 [error] Supervisor riak\_pipe\_vnode\_worker\_sup had child
undefined started with {riak\_pipe\_vnode\_worker,start\_link,undefined} at
<0.2681.0> exit with reason processing\_error in context child\_terminated
16:24:01.567 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,{kvget\_map,0},{trace,[error],{vnode,{fitting\_died,411047335499316445744786359201454599278231027712}}}}
16:24:01.622 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,{kvget\_map,0},{trace,[error],{vnode,{fitting\_died,22835963083295358096932575511191922182123945984}}}}
16:24:01.645 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,{kvget\_map,0},{trace,[error],{vnode,{fitting\_died,0}}}}
16:24:01.667 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,{kvget\_map,0},{trace,[error],{vnode,{fitting\_died,1392993748081016843912887106182707253109560705024}}}}
16:24:01.729 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.8144>,listkeys,{trace,[error],{vnode,{fitting\_died,707914855582156101004909840846949587645842325504}}}}



get\_map\_bucket\_values {} Record=rRecord, Nodes=["jason1"], CategoryList=[]
get\_map\_bucket\_values {} Record=r1Record, Nodes=["jason1"], CategoryList=[]
18:30:48.630 [error] gen\_fsm <0.9097.0> in state wait\_for\_input terminated
with reason: processing\_error
18:30:48.661 [error] CRASH REPORT Process <0.9097.0> with 0 neighbours
exited with reason: processing\_error in gen\_fsm:terminate/7
18:30:48.690 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,22835963083295358096932575511191922182123945984}}}}
18:30:48.717 [error] Supervisor riak\_pipe\_vnode\_worker\_sup had child
undefined started with {riak\_pipe\_vnode\_worker,start\_link,undefined} at
<0.9097.0> exit with reason processing\_error in context child\_terminated
18:30:48.750 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,936274486415109681974235595958868809467081785344}}}}
18:30:48.772 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,1392993748081016843912887106182707253109560705024}}}}
18:30:48.794 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,1255977969581244695331291653115555720016817029120}}}}
18:30:48.817 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,616571003248974668617179538802181898917346541568}}}}
18:30:48.828 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,0}}}}
18:30:48.839 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,411047335499316445744786359201454599278231027712}}}}
18:30:48.850 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,799258707915337533392640142891717276374338109440}}}}
18:30:48.872 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,685078892498860742907977265335757665463718379520}}}}
18:30:48.986 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,1187470080331358621040493926581979953470445191168}}}}
18:30:49.006 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,707914855582156101004909840846949587645842325504}}}}
18:30:49.039 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,{xform\_map,0},{trace,[error],{vnode,{fitting\_died,662242929415565384811044689824565743281594433536}}}}
18:30:49.072 [error] Unrecognized message
{pipe\_log,#Ref<0.0.0.63402>,1,{trace,[error],{vnode,{fitting\_died,45671926166590716193865151022383844364247891968}}}}

-- 
只为成功找方法，不为失败找理由
\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_\_
riak-users mailing list
riak-users@lists.basho.com
http://lists.basho.com/mailman/listinfo/riak-users\_lists.basho.com

